kythe/go/platform/kzip/kzip.go

```diff
@@ -95,6 +97,46 @@ import (
 	_ "kythe.io/kythe/proto/java_go_proto"
 )
 
+// Type Encoding describes how compilation units will be encoded when written to a kzip.
+type Encoding int32
+
+const (
+	BOTH  Encoding = 0
+	JSON  Encoding = 1
+	PROTO Encoding = 2
+)
+
+// Set updates an Encoding based on the text value
+func (e *Encoding) Set(v string) error {
+	switch {
+	case v == "BOTH":
+		*e = BOTH
+		return nil
+	case v == "JSON":
+		*e = JSON
+		return nil
+	case v == "PROTO":
+		*e = PROTO
+		return nil
+	default:
+		return fmt.Errorf("Unknown encoding %s", e)
+	}
+}
+
+// String stringifies an Encoding
+func (e Encoding) String() string {
+	switch {
+	case e == BOTH:
+		return "BOTH"
+	case e == JSON:
+		return "JSON"
+	case e == PROTO:
+		return "PROTO"
+	default:
+		return "Encoding" + strconv.FormatInt(int64(e), 10)
```

Comment:
`return fmt.Sprintf("Encoding%d", int(e))`

Or just panic.

Reply:
nit: ---/\

Reply:
Prefer not to panic if someone is just printing something out (e.g. debugging spew)